### ISSUE TITLE
`DateTime.diff` produces different `days` value if `months` is present

### ISSUE THREAD

**Issue by scotttrinh** [moment/luxon#487]
Simple reproduction case. Maybe it's related to March 31st -> April 1st? Or the quarter changing?

```ts
const { DateTime } = require("luxon");

const left = DateTime.fromJSDate(new Date(1554036127038));
const right = DateTime.fromJSDate(new Date(1554122527128));

const diff1 = right.diff(left, ["months", "days", "hours"]);
const diff2 = right.diff(left, ["days", "hours"]);

console.log(diff1.days); // Prints 2
console.log(diff2.days); // Prints 1
```

**Comment by icambron** [comment]
hmm, yeah, looks like a bug

**Comment by icambron** [comment]
Fixed in 6a926b2, will release soon.
